frontend/dockerfile/parser/errors.go

```diff
@@ -1,6 +1,10 @@
 package parser
 
 import (
+	"math"
+	"sort"
+
+	"github.com/moby/buildkit/solver/pb"
```

Comment:
Let's avoid adding this new dependency to the AST parser package.

`MergeLocations` also looks weird as a public API for the parser package. Options here are to make it a method of `ErrorLocations` (eg. `errLoc.Append(ranges)`) or to make it a private helper function in the caller side.